x/registry/module/module.go

```diff
@@ -0,0 +1,106 @@
+package module
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	client "github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/types/msgservice"
+
+	"github.com/provenance-io/provenance/x/registry/client/cli"
+	"github.com/provenance-io/provenance/x/registry/keeper"
+	registrytypes "github.com/provenance-io/provenance/x/registry/types"
+)
+
+var _ module.AppModule = AppModule{}
+var _ module.AppModuleBasic = AppModuleBasic{}
+
+type AppModuleBasic struct {
+	cdc codec.Codec
+}
+
+// Name returns the module name.
+func (AppModuleBasic) Name() string {
+	return registrytypes.ModuleName
+}
+
+// GetTxCmd returns the root tx command for the registry module.
+func (AppModuleBasic) GetTxCmd() *cobra.Command {
+	return cli.CmdTx()
+}
+
+func (AppModuleBasic) GetQueryCmd() *cobra.Command {
+	return cli.CmdQuery()
+}
+
+// RegisterCodec registers the module's types.
+func (AppModuleBasic) RegisterCodec(_ *codec.LegacyAmino) {
+	// Register any concrete types if needed.
+}
+
+func (AppModuleBasic) RegisterLegacyAminoCodec(*codec.LegacyAmino) {}
+
+// Register the protobuf message types and services with the sdk.
+func (AppModuleBasic) RegisterInterfaces(r types.InterfaceRegistry) {
+	msgservice.RegisterMsgServiceDesc(r, &registrytypes.Msg_serviceDesc)
```

Comment:
issue (blocking): This needs to be a call to a function in `types/codec.go` (similar to the other modules).